gui/src/app/state/receive.rs

```diff
@@ -154,12 +155,20 @@ impl State for ReceivePanel {
                 ));
                 Command::none()
             }
+            Message::View(view::Message::ShowQrCode(i)) => {
+                let address = self.addresses.list.get(i).expect("Must be present").clone();
+                let qr_code = qr_code::State::new(format!("bitcoin:{}?index={}", address, i)).ok();
```

Comment:
I would move this code in the `ShowQrCodeModal::new(address, index) -> Option<ShowQrCodeModal>`
and do `self.modal = ShowQrCodeModal::new(address, i)`